### ISSUE TITLE
Add deprecation notice

### ISSUE THREAD

**Issue by TehShrike** [coding-in-the-wild/just-login-example-session-manager#3]
Session expiration is handled by the optional [just-login-session-state](https://github.com/coding-in-the-wild/just-login-session-state), and doesn't need to be in this module.

Related: should this module be renamed to `just-login-session-manager` or maybe `just-login-session-creator`?  Maybe we should rethink this functionality as a new module entirely?

**Comment by ArtskydJ** [comment]
Agreed on removing session expiration from this module.

This module works well with RPC (like dnode), but does it work well with events (like socket.io for example)?

**Comment by TehShrike** [comment]
Maybe?  It would probably look like:

``` js
var manager = new SessionManager(core, db)

io.on('connection', function(socket) {
    socket.on('give me a session', function() {
        manager.createSession(function(err, managerApi, sessionId) {
            handleSocketWithSession(socket, managerApi)
        })
    })

    socket.on('continue session', function(sessionId) {
        manager.continueSession(sessionId, function(err, managerApi, sessionId) {
            handleSocketWithSession(socket, managerApi)
        })
    })
})

function handleSocketWithSession(socket, managerApi) {
    socket.on('thing', function() {
        managerApi.isAuthenticated(function(err, emailAddress) {
            database.doStuff()
        })
    })

    socket.on('login', function(emailAddress) {
        managerApi.beginAuthentication(emailAddress)
    })

    socket.on('logout', function() {
        managerApi.unauthenticate()
    })
}
```

But I don't know if there's that much advantage to doing that than passing in the session id instead of the managerApi to handleSocketWithSession.  Or, for that matter, building a similar helper around using JWT instead.

**Comment by ArtskydJ** [comment]
After [`just-login-session-state` #5](https://github.com/coding-in-the-wild/just-login-session-state/pull/5), this module is can be reduced to:

``` js
module.exports = function (jlc, sessionId) {
    return {
        beginAuthentication: jlc.beginAuthentication.bind(jlc, sessionId),
        isAuthenticated: jlc.isAuthenticated.bind(jlc, sessionId),
        unauthenticate: jlc.unauthenticate.bind(jlc, sessionId)
    }
}
```

At that point, is it worth keeping this?

**Comment by TehShrike** [comment]
It still handles creating new sessions and determining whether or not sessions existed already, right?

**Comment by ArtskydJ** [comment]
I felt like `createSession()` and `continueSession()` made sense with `just-login-session-state`.

[See PR](https://github.com/coding-in-the-wild/just-login-session-state/pull/5/files) (scroll halfway down).

What do you think?

**Comment by TehShrike** [comment]
I'd missed that pull request apparently.  Yeah, if something like https://github.com/coding-in-the-wild/just-login-session-state/pull/5 goes in, this module isn't needed any more at all as far as I can tell.

**Comment by ArtskydJ** [comment]
Agreed.

**Comment by TehShrike** [comment]
I suppose at this point we can change the readme to say something about how this module is deprecated, and link to just-login-session-state.

**Comment by ArtskydJ** [comment]
Fixed in 86ef24d5fb076ca2ae44487b4033d60f5299069a.

Published as v1.2.1 (so that the deprecation notice is viewable on the npm website.)
